src/store/store.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useEffect } from 'react'
+import { useNavigate } from 'react-router'
+import create, { SetState, State, StateCreator } from 'zustand'
+import createContext from 'zustand/context'
+import { persist } from 'zustand/middleware'
+
+import { SnackbarState, useSnackbar } from './snackbars'
+import { UploadStoreState, useUploadsManagerStore } from './uploads'
+
+const ZUSTAND_LOCAL_STORAGE_KEY = 'uploads_zustand'
+
+const { useStore, Provider } = createContext<RootState>()
+
+export type RootState = SnackbarState & UploadStoreState
+
+export const combineStateCreators = <TState extends State>(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ...stateCreators: StateCreator<any>[]
+): StateCreator<TState> => (set, get, api) => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let values: any = {}
+
+  stateCreators.forEach((sc) => {
+    values = Object.assign({}, values, sc(set, get, api))
+  })
+  return values
+}
```

Comment:
found here https://codesandbox.io/s/zustand-combine-state-creators-y0eev?file=/src/zustand/middleware.ts

Reply:
Probably we should in the end split the store(for snack bars and for uploads). 

I'll just leave it to show that creating a root state is possible, and it's also a little bit awkward